src/google/protobuf/stubs/logging.h

```diff
@@ -59,6 +59,19 @@ enum LogLevel {
 #else
   LOGLEVEL_DFATAL = LOGLEVEL_FATAL
 #endif
+
+#ifdef _WIN32
+  // wingdi.h defines ERROR to be 0, so `GOOGLE_LOG(ERROR, ...)` expands
+  // into `GOOGLE_LOG(0, ...)` which then expands into
+  // `someGoogleLogging(LOGLEVEL_0, ...)`. This is not ideal, because the
+  // GOOGLE_LOG macro expects to expand itself into
+  // `someGoogleLogging(LOGLEVEL_ERROR, ...)` instead. The workaround to
+  // get everything building is to simply define LOGLEVEL_0 as
+  // LOGLEVEL_ERROR and also define ERROR the same way that the Windows
+  // SDK does for consistency.
+#define ERROR 0
+  , LOGLEVEL_0 = LOGLEVEL_ERROR
```

Comment:
How does wingdi.h get included in protobuf code?

Reply:
It gets included via `Windows.h`.

Reply:
Would defining NOGDI work for your project?

Reply:
Thanks for that idea. I defined `NOGDI` in our build files and it now works fine without this patch.

That seems like a much better work-around. If the logging system is going to be replaced, then there's not much point in merging this pull request.